sources/presentation/Stride.Core.Quantum/GraphNodeBase.cs

```diff
@@ -69,7 +70,21 @@ public static IEnumerable<NodeIndex> GetIndices([NotNull] IGraphNode node)
         {
             if (node.Descriptor is CollectionDescriptor collectionDescriptor)
             {
-                return Enumerable.Range(0, collectionDescriptor.GetCollectionCount(node.Retrieve())).Select(x => new NodeIndex(x));
+                if (node.Descriptor.Category == DescriptorCategory.Set)
+                {
+                    var enumerator = (node.Retrieve() as IEnumerable).GetEnumerator();
+                    NodeIndex[] valueArr = new NodeIndex[collectionDescriptor.GetCollectionCount(node.Retrieve())];
+                    int i = 0;
+                    while (enumerator.MoveNext())
+                    {
+                        valueArr[i++] = new NodeIndex(enumerator.Current);
+                    }
+                    return valueArr;
+                }
+                else
+                {
+                    return Enumerable.Range(0, collectionDescriptor.GetCollectionCount(node.Retrieve())).Select(x => new NodeIndex(x));
+                }
```

Comment:
For ISet, we shouldn't use int index like IList to build it's NodeIndex in editor. The index of ISet should be the element value itself to find the element.
So the logic here is nessary, or it may cause getting wrong element when deleting or changing element in ISet.